src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolverInterface.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+/**
+ * Responsible for the value of an argument based on its metadata.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+interface ArgumentValueResolverInterface
+{
+    /**
+     * Should return true if this resolver can resolve the value for the given ArgumentMetadata.
+     *
+     * @param Request          $request
+     * @param ArgumentMetadata $argument
+     *
+     * @return bool
+     */
+    public function supports(Request $request, ArgumentMetadata $argument);
+
+    /**
+     * Yield the possible value(s).
+     *
+     * An implementation must yield at least one value.
```

Comment:
@HeahDude  no, because the value resolver won't be responsible for throwing here (it can throw if there is an error, but this is not something being part of the interface here, it is just a normal way to report errors in PHP).

Returning an empty iterator would be a broken implementation. and the ArgumentResolver may throw an exception in this case to help debugging it (but it does not currently, it just let the resolving become a mess)